tswow-core/Public/TSItemTemplate.h

```diff
@@ -23,42 +23,56 @@ struct ItemTemplate;
 
 class TC_GAME_API TSItemTemplate : public TSEntityProvider {
 private:
-    ItemTemplate * info;
+    ItemTemplate* info;
 public:
     bool IsNull() { return info == nullptr; };
     TSItemTemplate(ItemTemplate const* info);
     TSItemTemplate();
-    TSItemTemplate* operator->() { return this;}
+    TSItemTemplate* operator->() { return this; }
     operator bool() const { return info != nullptr; }
     bool operator==(TSItemTemplate const& rhs) { return info == rhs.info; }
 
     uint32 GetEntry();
 
+    void SetEntry(uint32 value);
+
     float GetDamageMinA();
+    void SetDamageMinA(float value);
     float GetDamageMinB();
-
+    void SetDamageMinB(float value);
     float GetDamageMaxA();
+    void SetDamageMaxA(float value);
     float GetDamageMaxB();
-
+    void SetDamageMaxB(float value);
     uint32 GetDamageTypeA();
+    void SetDamageTypeA(uint32 value);
     uint32 GetDamageTypeB();
+    void SetDamageTypeB(uint32 value);
 
     uint32 GetClass();
+    void SetClass(uint32 value);
     uint32 GetSubClass();
+    void SetSubClass(uint32 value);
     int32  GetSoundOverrideSubclass();
     TSString GetName();
+    void SetName(TSString name);
     uint32 GetDisplayInfoID();
+    void SetDisplayInfoID(uint32 value);
     uint32 GetQuality();
+    void SetQuality(uint32 value);
```

Comment:
If modified it will modify the base item itself(unless using the copy function), as this was setup to allow for blizzlike item changes/reloads along with custom, it was left on the actual item_template

Reply:
All of these functions could be moved to the virtual item on the other review message so it's only accessible from there, this just allows changing a blizzlike and reloading it to apply to all items.

Reply:
If that reload works reliably I don't mind this being available for normal item templates as well

Reply:
I have not had an issue, but there is some live script boilerplate to make it work 'seamlessly'

Reply:
tested spells and worked fine, so i added all of the fields and worst case, amend later.